pandas/core/index.py

```diff
@@ -1179,17 +1179,18 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if isinstance(other, Index):
+        if com.is_list_like(other):
             warnings.warn("using '+' to provide set union with Indexes is deprecated, "
                           "use '|' or .union()",FutureWarning)
+        if isinstance(other, Index):
```

Comment:
this is not correct I think, as now a list will show the warning, but not perform a set (union) operation (as a list is not an index?

Should it not be like:

```
if com.is_list_like(other):
    warnings.warn(...)
    return self.union(other)
```